src/Agent/NewRelic/Agent/Core/Api/AgentApi.cs

```diff
@@ -280,6 +281,22 @@ void work()
             TryInvoke(work, apiName, apiMetric);
         }
 
+        /// <summary>
+        /// Notice an error identified by a simple message and report it to the New Relic service.
+        /// If this method is called within a transaction,
+        /// the exception will be reported with the transaction when it finishes.  
+        /// If it is invoked outside of a transaction, a traced error will be created and reported to the New Relic service.
+        /// Only the string/parameter pair for the first call to NoticeError during the course of a transaction is retained.
+        /// Supports web applications only. 
```

Comment:
Woah, what? Why would this only support Web applications? Does this mean that if this is called within any custom non-web instrumentation that the error is not actually Noticed?

Reply:
I don't think the API call has any detection capability, the message seems odd.